src/v/datalake/record_schema_resolver.cc

```diff
@@ -221,6 +202,44 @@ binary_type_resolver::resolve_identifier(schema_identifier) const {
     co_return type_resolver::errc::translation_error;
 }
 
+ss::future<checked<shared_schema_t, type_resolver::errc>>
+record_schema_resolver::get_schema(ppsr::schema_id id) const {
+    if (!sr_.is_enabled()) {
+        vlog(datalake_log.warn, "Schema registry is not enabled");
+        // TODO: should we treat this as transient?
+        co_return type_resolver::errc::translation_error;
+    }
+    if (cache_.has_value()) {
+        auto cached_schema = cache_->get().get_value(id);
+
+        if (cached_schema) {
+            co_return std::move(*cached_schema);
+        }
```

Comment:
nit: would be great to have some debug/trace log lines here to help understand when we actually access SR. And/or a metric tick, though maybe the metric tick is better placed in the cache itself

Reply:
Great idea, let me try adding hit/miss metrics to the cache and that optionally be enabled.